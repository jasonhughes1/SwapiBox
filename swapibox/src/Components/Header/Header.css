.header {
  background-color: black;
  background-image: url("http://localhost:3000/images/yoda.jpg");
  background-repeat: no-repeat;
  height: 480px; }

.main-header {
  color: yellow;
  background-color: black; }

.fav-button {
  background-color: yellow; }
