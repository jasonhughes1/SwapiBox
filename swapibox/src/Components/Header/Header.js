import React from 'react';
import css from './Header.css';
import Controls from './../Controls/Controls.js'


const Header = ({favFn, numFav, changeCards, num}) => {
  return (
    <div className="header-container">
      <h1 className="main-header">SWAPI-Box</h1>
      <Controls className={' button favorite '}
               buttonText={'View Favorites' + numFav}
                    favFn={favFn}
              changeCards={changeCards}
              num={num}
      />
    </div>
  )
}


export default Header;
