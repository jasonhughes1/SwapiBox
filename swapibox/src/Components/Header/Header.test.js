import React from 'react';
import Header from './Header.js';
import { shallow } from 'enzyme';


describe('Header tests', () => {


  it('should match snapshot()', () => {
    let renderedApp = shallow(<Header />);

    expect(renderedApp).toMatchSnapshot();
  });

  describe('Header', () => {
  it('should render a header', () => {
    const renderedHeader = shallow(<Header />);
    expect(renderedHeader.find('.header-container').length).toEqual(1);
  });
});
});
