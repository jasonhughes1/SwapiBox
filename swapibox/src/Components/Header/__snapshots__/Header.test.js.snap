// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should match snapshot() 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-header"
>
          SWAPI-Box
</h1>,
        <div
          className="header"
/>,
        <Controls
          buttonText="View Favorites:  undefined"
          changeCards={undefined}
          className=" button-favorite "
          favFn={undefined}
          num={undefined}
/>,
      ],
      "className": "header-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI-Box",
          "className": "main-header",
        },
        "ref": null,
        "rendered": "SWAPI-Box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "header",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonText": "View Favorites:  undefined",
          "changeCards": undefined,
          "className": " button-favorite ",
          "favFn": undefined,
          "num": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-header"
>
            SWAPI-Box
</h1>,
          <div
            className="header"
/>,
          <Controls
            buttonText="View Favorites:  undefined"
            changeCards={undefined}
            className=" button-favorite "
            favFn={undefined}
            num={undefined}
/>,
        ],
        "className": "header-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI-Box",
            "className": "main-header",
          },
          "ref": null,
          "rendered": "SWAPI-Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "header",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "View Favorites:  undefined",
            "changeCards": undefined,
            "className": " button-favorite ",
            "favFn": undefined,
            "num": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
